src/api/test-controller.js

```diff
@@ -107,10 +109,9 @@ export default class TestController {
             throw err;
         }
 
-        this.executionChain       = this.executionChain.then(() => this.testRun.executeCommand(command, callsite));
         this.callsiteWithoutAwait = callsite;
 
-        return this._createExtendedPromise(this.executionChain, callsite);
+        return this._createExtendedPromise(this.testRun.executeCommand(command, callsite), callsite);
```

Comment:
Well, it's not correct and completely kills any possibility for the error recovery.